docs(product tours): add localization docs (#15078)
PostHog/posthog.com: contents/docs/product-tours/localization.mdx

---
title: Localization
sidebar: Docs
showTitle: true
---

> import Alpha from './_snippets/alpha.mdx'

Reviewer: ```suggestion
import Alpha from './_snippets/alpha.mdx'
import { CalloutBox } from 'components/Docs/CalloutBox'
```

## Creating translations

### Set the default language

First, you'll need to set the default language for your tour.

When you set a default language, all existing tour content will be mapped to the new default.

This will be used as a fallback when the user's language cannot be determined, or your tour does not support the user's provided language.

Open the **Translations** panel in the editor, and enter a language code that represents your tour's default.

<ProductScreenshot
    imageLight="https://res.cloudinary.com/dmukukwp6/image/upload/Screenshot_2026_02_16_at_7_45_43_PM_c55596a1e6.png"
    alt="Product Tour Translations panel"
    classes="rounded"
/>

### Add a new language

Once the default is set, you can add more languages in the same **Translations** panel.

Enter a new language code, click the plus button to add it, then click the language card to select it.

<ProductScreenshot
    imageLight="https://res.cloudinary.com/dmukukwp6/image/upload/Screenshot_2026_02_16_at_8_04_07_PM_53f2b73cdc.png"
    alt="Product Tour Translations panel"
    classes="rounded"
/>

> Once selected, you'll see a new language code at the top of the editor, and you can make any changes necessary for this translation.

Reviewer 2: ```suggestion
Once selected, a new language code appears at the top of the editor, and you can make any changes necessary for this translation.
```

If you have other translatable fields beyond the pop-up content, they will appear below the content editor.

<ProductScreenshot
    imageLight="https://res.cloudinary.com/dmukukwp6/image/upload/Screenshot_2026_02_16_at_8_06_51_PM_f9ac90017b.png"
    alt="Product Tour Translations panel"
    classes="rounded"
/>

## Language code matching

When determining which tour translation to use, we run the following checks in this order:

1. **Exact match**: a user with language `en-US` matches a tour with translation `en-US`
2. **Base match**: if no exact matches are found, we check for translations matching the user's "base" language. For example, a user language `en-GB` will match a tour with translation `en`, if no translation exists specifically for `en-GB`.
3. **Fallback**: if neither of the above are successful, we fall back to the tour's default language.

> **Note:** the base matching only works one way; a user with language `en` will not match a tour translation `en-US`.
> 
> For this reason we recommend keeping your translation keys as base-only (like `en`), unless you have translations for specific dialects.

## Language overrides

> By default, we'll use the user's browser language to determine which tour translation to show.

Reviewer 2: ```suggestion
By default, PostHog uses the user's browser language to determine which tour translation to show.
```

This can be overridden by setting the `override_display_language` property in `posthog.init`.

```typescript
// set in posthog.init
posthog.init('api-key', {
  api_host: 'api-host',
  defaults: '2025-05-24',
  disable_product_tours: false,
  override_display_language: 'en-US',
});
```

You can also use the `posthog.set_config` method to change the configuration after initialization.

```typescript
posthog.set_config({
  override_display_language: 'en-GB',
})
```

## Tips & tricks

> ### Which languages should you support?

Reviewer 2: No need for an `FAQ` or `Tips & tricks` section that's only one entry long. We can add it back if this section grows longer

```suggestion

## Which languages should you support?
```

> ### Which languages should you support?

Reviewer: ```suggestion
## Find what languages to support
```
